src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -2248,6 +2248,10 @@ bool Creature::CanCreatureAttack(Unit const* victim, bool skipDistCheck) const
     if (victim->GetTypeId() == TYPEID_UNIT && victim->ToCreature()->IsInEvadeMode())
         return false;
 
+    // cannot attack if is during 5 second grace period, unless being attacked
+    if ((time(nullptr) - m_respawnedTime) < 5 && victim->getAttackers().empty())
+        return false;
```

Comment:
@WojtekNosal please add brackets, this helps preventing to introduce issues by mistake later ( sounds strict but I've seen it happening :D )

Reply:
@FrancescoBorzi done! :D